python/pyglome/glome.py

```diff
@@ -217,3 +217,143 @@ def generate_keys() -> KeyPair:
+
+    @property
+    def receiving_counter(self) -> int:
+        """Number of tags the user receives from the peer.
+
+        It is incremented each time a new tag is checked. It is
+        always one byte long. When the counter gets past 255
+        it restarts at 0.
+
+        Setter raises ValueError if provided counter is not in range 0-255.
+        """
+        return self._receiving_counter
+
+    @receiving_counter.setter
+    def receiving_counter(self, value: int):
+        if not 0 <= value <= 255:
+            raise ValueError('Counter must be in range 0-255')
+        self._receiving_counter = value
+
+    @property
+    def user_keys(self) -> KeyPair:
+        """User's private and public keys used in handshake."""
+        return self.glome.user_keys
+
+    @property
+    def peer_key(self) -> x25519.X25519PublicKey:
+        """Peer's public key used in handshake."""
+        return self.glome.peer_key
+
+    def tag(self, msg: bytes) -> bytes:
+        """Generates a tag from a message.
+
+        Generates a tag matching some provided message and the internal
+        sending counter. This tag is generated following GLOME protocol
+        specification in the context of a communication from the users to
+        theirs peers.
+
+        Args:
+           msg: Message to be transmitted.
+        Returns:
+           tag: Tag matching counter and msg.
+        Raises:
+           TagGenerationError: Raised whenever the method failed to generate tag
+             due to ValueError in the arguments.
+        """
+        tag = self.glome.tag(msg, self.sending_counter)
+        self._sending_counter = (self._sending_counter + 1) % 256
+        return tag
+
+    def check(self, tag: bytes, msg: bytes):
+        """Check whether a tag is correct for some message.
+
+        Checks if a tag matches some provided message and internal receiving
+        counter. The method generates the matching tag following GLOME protocol
+        specification in the context of a communication from the users' peers to
+        the users and then is compared with the tag provided. If tag checking if 
+        not successful, the receiving counter remains unchanged.
+
+        If skippable_range if greater than 0, the method try to check the tag 
+        against all counters in range [receiving_counter, receiving_counter + 
+        skippable_range], in order, until one is successful. If no one is successful,
+        an exceptions is raised and receiving counter remains unchanged.
+   
+        Args:
+           tag: Object with the generated tag.
+           msg: Object containing received message.
+        Returns:
+           None.
+        Raises:
+           IncorrectTagError: Raised whenever the tag is incorrect.
+        """
+        old_counter = self._receiving_counter
+        for i in range(self.skippable_range + 1):
+            try:
+                self.glome.check(tag, msg, self.receiving_counter)
+                return None
```

Comment:
Should there be an increment of the counter here?

Reply:
Yes, it should be.

Reply:
fixed!